server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java

```diff
@@ -1426,6 +1435,53 @@ public void checkTemplateOnPath(PartialPath path) throws MetadataException {
     checkTemplateOnSubtree(cur);
   }
 
+  /**
+   * Check route 1: If template has no direct measurement, just pass the check.
+   *
+   * <p>Check route 2: If template has direct measurement and mounted node is Internal, it should be
+   * set to Entity.
+   *
+   * <p>Check route 3: If template has direct measurement and mounted node is Entity,
+   *
+   * <p>route 3.1: mounted node has no measurement child, then its alignment will be set as the
+   * template.
+   *
+   * <p>route 3.2: mounted node has measurement child, then alignment of it and template should be
+   * identical, otherwise cast a exception.
+   *
+   * @return return the node competent to be mounted.
+   */
+  public IMNode checkTemplateAlignmentWithMountedNode(IMNode mountedNode, Template template)
+      throws MetadataException {
+    boolean hasDirectMeasurement = false;
+    for (IMNode child : template.getDirectNodes()) {
+      if (child.isMeasurement()) {
+        hasDirectMeasurement = true;
+      }
+    }
+    if (hasDirectMeasurement) {
+      if (!mountedNode.isEntity()) {
+        return setToEntity(mountedNode);
+      } else {
+        for (IMNode child : mountedNode.getChildren().values()) {
+          if (child.isMeasurement()) {
+            if (template.getAlignedPrefixSet().contains("")
```

Comment:
What's the meaning of empty string here?

Reply:
Thanks! This string tries to represent alignment of direct measurement of a template. I will add more comment.